[Bug]: Test pipeline failure on windows

### Bug summary

https://dev.azure.com/matplotlib/matplotlib/_build/results?buildId=44174&view=logs&j=c0a65fc6-a4f7-5116-4cee-beb3104aaa7c&t=ccc2daa7-0687-5084-0ea1-0e98a3640679&l=239

```
        Return the number of significant digits to %g-format *value*, assuming that
        it is known with an error of *delta*.
        """
        if delta == 0:
            if value == 0:
                # if both value and delta are 0, np.spacing below returns 5e-324
                # which results in rather silly results
                return 3
            # delta = 0 may occur when trying to format values over a tiny range;
            # in that case, replace it by the distance to the closest float.
>           delta = abs(np.spacing(value))
                        ^^^^^^^^^^^^^^^^^
E           RuntimeWarning: invalid value encountered in spacing

lib\matplotlib\cbook.py:2238: RuntimeWarning
```

Possibly related to the numpy 2.3 release from yesterday.

### Code for reproduction

```Python
-
```

### Actual outcome

-

### Expected outcome

-

### Additional information

_No response_

### Operating system

_No response_

### Matplotlib Version

Main

### Matplotlib Backend

_No response_

### Python version

_No response_

### Jupyter version

_No response_

### Installation

None

--- a/lib/matplotlib/cbook.py
+++ b/lib/matplotlib/cbook.py
@@ -2228,6 +2228,9 @@ def _g_sig_digits(value, delta):
     Return the number of significant digits to %g-format *value*, assuming that
     it is known with an error of *delta*.
     """
+    # For inf or nan, the precision doesn't matter.
+    if not math.isfinite(value):
+        return 0
     if delta == 0:
         if value == 0:
             # if both value and delta are 0, np.spacing below returns 5e-324
@@ -2241,11 +2244,10 @@ def _g_sig_digits(value, delta):
     # digits before the decimal point (floor(log10(45.67)) + 1 = 2): the total
     # is 4 significant digits.  A value of 0 contributes 1 "digit" before the
     # decimal point.
-    # For inf or nan, the precision doesn't matter.
     return max(
         0,
         (math.floor(math.log10(abs(value))) + 1 if value else 1)
-        - math.floor(math.log10(delta))) if math.isfinite(value) else 0
+        - math.floor(math.log10(delta)))
 
 
 def _unikey_or_keysym_to_mplkey(unikey, keysym):

